### ISSUE TITLE
yapf only on emacs not support characters other than ascii.

### ISSUE THREAD

**Issue by myme5261314** [syl20bnr/spacemacs#7429]
#### Description :octocat:

yapf only on emacs not support characters other than ascii.
#### Reproduction guide :beetle:
- Start Emacs
- use python snippets

``` python
#! /usr/bin/env python
# -*- coding: utf-8 -*-
#
# Copyright ©  2016
# XXX XXX, <XXX@gmail.com>
#
# Distributed under terms of the gplv3 license.
def main():
    pass
if __name__ == '__main__':
    main()
# Notice the character © which is \\xa9. 
```
- press `, =`

_Observed behaviour:_ :eyes: :broken_heart:

```
"Yapf failed with the following error(s): 

Traceback (most recent call last):
  File \"/usr/local/bin/yapf\", line 11, in <module>
    sys.exit(run_main())
  File \"/usr/local/lib/python2.7/dist-packages/yapf/__init__.py\", line 259, in run_main
    sys.exit(main(sys.argv))
  File \"/usr/local/lib/python2.7/dist-packages/yapf/__init__.py\", line 163, in main
    sys.stdout.write(reformatted_source)

UnicodeEncodeError: 'ascii' codec can't encode character u'\\xa9' in position 61: ordinal not in range(128)
"
```

_Expected behaviour:_ :heart: :smile:
No error.
Use shell to run `yapf -i test.py` will not produce error.
#### System Info :computer:
- OS: gnu/linux
- Emacs: 24.5.1
- Spacemacs: 0.200.2
- Spacemacs branch: develop (rev. 6cb349e)
- Graphic display: t
- Distribution: spacemacs
- Editing style: vim
- Completion: helm
- Layers:

``` elisp
(better-defaults github
                 (version-control :variables version-control-diff-tool 'git-gutter  version-control-global-margin t)
                 semantic markdown
                 (vinegar :variables vinegar-reuse-dired-buffer t)
                 org prodigy search-engine
                 (syntax-checking :variables syntax-checking-enable-by-default nil)
                 (spell-checking :variables spell-checking-enable-by-default nil)
                 yaml python lua html command-log emacs-lisp emoji fasd deft ranger gtags
                 (spacemacs-layouts :variables layouts-enable-autosave t layouts-autosave-delay 300)
                 (colors :variables colors-enable-nyan-cat-progress-bar t)
                 (git :variables git-magit-status-fullscreen t magit-push-always-verify nil magit-save-repository-buffers 'dontask magit-revert-buffers 'silent magit-refs-show-commit-count 'all magit-revision-show-gravatars nil)
                 (ibuffer :variables ibuffer-group-buffers-by 'projects)
                 (c-c   :variables c-c  -default-mode-for-headers 'c  -mode)
                 (auto-completion :variables auto-completion-enable-sort-by-usage t auto-completion-enable-snippets-in-popup t)
                 (shell :variables shell-default-position 'full shell-default-shell 'multi-term shell-default-term-shell "/bin/zsh")
                 (chinese :variables chinese-default-input-method 'pinyin chinese-enable-fcitx t chinese-enable-youdao-dict t)
                 emacs-lisp markdown org dash bibtex django pandoc my)
```
#### Backtrace :paw_prints:

```
<<BACKTRACE IF RELEVANT>>
```

**Comment by myme5261314** [comment]
I don't think this should be labeled as `Should report upstream` since using yapf within shell succeed.

**Comment by myme5261314** [comment]
Ok, maybe you mean the package `yapf.el`? My apology.

**Comment by robbyoconnor** [comment]
Yes, this should be upstream. We don't do anything special here.

**Comment by TheBB** [comment]
We use yapfify, not py-yapf. Please report upstream to the right package. :smile:

**Comment by TheBB** [comment]
For the record I can't reproduce this, it works fine here.

**Comment by myme5261314** [comment]
@TheBB you sure you can't reproduce this?
I test out that adding any characters other than ascii (such as chinese) will make the error occur.
Remove the special character will made no error.
Any idea about how to get more diagnostic info?

**Comment by TheBB** [comment]
Yes, I ran `,=` on your sample code; no errors. Perhaps your locale isn't set to UTF-8?

**Comment by myme5261314** [comment]
on my zsh shell
```shell
>> echo $LANG
en_US.UTF-8
```

on the spacemacs shell cmd `SPACE ! `
```shell
>> echo $LANG
zh_CN.UTF-8
```

And my sample code contains the utf-8 header.
```python
#! /usr/bin/env python
# -*- coding: utf-8 -*-
```

Can't see any weird settings.

**Comment by myme5261314** [comment]
Seems have been fixed in more recent version of yapf package.
Confirmed bug in yapf 0.11.1, confirmed bug resolved in 0.16.2.
Closing.
